rc/tools/git.kak

```diff
@@ -71,22 +77,27 @@ define-command -params 1.. \
     if [ $kak_token_to_complete -eq 0 ]; then
         printf "add\nrm\nreset\nblame\ncommit\ncheckout\ndiff\nhide-blame\nhide-diff\nlog\nnext-hunk\nprev-hunk\nshow\nshow-branch\nshow-diff\ninit\nstatus\nupdate-diff\ngrep\n"
     else
+        # Try cd-ing into the git work tree if exists to make autocompletion slightly more robust
+        cd "$kak_opt_git_work_tree_top_level" >/dev/null 2>&1
         case "$1" in
             commit) printf -- "--amend\n--no-edit\n--all\n--reset-author\n--fixup\n--squash\n"; git ls-files -m ;;
             add) git ls-files -dmo --exclude-standard ;;
             rm|grep) git ls-files -c ;;
         esac
     fi
   } \
-  git %{ evaluate-commands %sh{
-    cd_bufdir() {
-        dirname_buffer="${kak_buffile%/*}"
-        cd "${dirname_buffer}" 2>/dev/null || {
-            printf 'fail Unable to change the current working directory to: %s\n' "${dirname_buffer}"
-            exit 1
-        }
-    }
-
+  git %{
+  set-option global git_work_tree_top_level %sh{
```

Comment:
It doesn't seem right to set this in the global scope (it's a source of wrong behavior in completion and `git-diff-goto-source`).
Ideally we get rid of this option, it appears though only `git-diff-goto-source` needs it.
I see two potential fixes:
1. set the option only at buffer scope for buffers where we bind `git-diff-goto-source`
2. bypass  `git-diff-goto-source` and encode the git directory where we map it

		"unmap buffer normal <ret> :diff-jump $PWD<ret>"

I prefer 2 because then we don't need a new option.

Reply:
Not sure if I follow the 2 variant. What is on your mind exactly? If you remember after the month 🙈

Reply:
I see the appeal of the variant 2. but at the same time having the option is really handy for any external scripting that would like to be "reasonable git-work-tree" aware.